### DISCUSSION TITLE
Valet custom driver does not load css correctly

### DISCUSSION THREAD

**Discussion by henrik561** [laravel/valet#1492] [Q&A]
I have a wordpress site with an .htaccess in it, but when i tried to load the project i had no styles so i did some research and i found out that valet runs on nginx and .htaccess only works with apache so i had to write a custom valet driver, so that is what i did and i got the following:
```
<?php

namespace Valet\Drivers\Custom;

use Valet\Drivers\ValetDriver;

class WordpressValetDriver extends ValetDriver
{
    public function serves(string $sitePath, string $siteName, string $uri): bool
    {
        return $siteName == 'houtje-touwtje';
    }

    /**
     * Determine if the incoming request is for a static file.
     */
    public function isStaticFile(string $sitePath, string $siteName, string $uri)
    {
        // if (file_exists($staticFilePath = $sitePath . '/public/' . $uri)) {
        //     return $staticFilePath;
        // }
        return false;
    }

    /**
     * Get the fully resolved path to the application's front controller.
     */
    public function frontControllerPath(string $sitePath, string $siteName, string $uri): string
    {
        $indexPath = $sitePath . '/index.php';



        $rules = [
            [
                'pattern' => '/^\/static\/ext\/(.*)$/',
                'replacement' => '/wp-content/plugins/$1',
            ],
            // there are more rules but those do not add any value to this question
        ];

        foreach ($rules as $rule) {
            if (preg_match($rule['pattern'], $uri)) {
                if ($rule['replacement'] !== null) {
                    $uri = preg_replace($rule['pattern'], $rule['replacement'], $uri);
               }

                return $sitePath . $uri;
            }
        }

        return $indexPath;
    }
}
```

and with some testing i got it to work, when i check my devtools i see the following:

![image](https://github.com/user-attachments/assets/889c1e77-e265-473d-bc12-3b656ad9ceba)
![image](https://github.com/user-attachments/assets/d7e72b51-3008-4cf7-be30-74cc0e24acec)

And the response is:
![image](https://github.com/user-attachments/assets/9144c68f-a2ee-41ec-845c-36cb03916f1b)
![image](https://github.com/user-attachments/assets/7b609454-0226-485c-ba57-34aaa22d3d11)

And when i visit the url for the flatsome.css file i get as response the css, but the weird thing is that the css is not being applied, so no where on my site is any css,

But images etc are all loaded fine as u can see in the image below
![image](https://github.com/user-attachments/assets/da15e6aa-0e4d-4db0-9657-29eb55fa8d2d)

**Environment**
php:  7.4
valet: 4.6.1

If u need any more details feel free to ask, and any help or suggestions are welcome

**Comment by drbyte** [comment]
> a wordpress site with an .htaccess in it
> valet runs on nginx and .htaccess only works with apache

What's the content of your `.htaccess` file?
It may be necessary to incorporate more of the rules it's providing, either into your custom driver, or into a custom nginx site-config.

**Reply by henrik561** [reply]
Thanks for your response, 

Sadly i was on vacation but it was due to the headers not being set to text/css for example, all the files were received as plain text.

But due to more issues decided to install MAMP for an easier fix instead of modifying my custom driver for 3 hours 😂
